cmd/proxy/actions/basicauth_test.go

```diff
@@ -1,57 +1,80 @@
 package actions
 
 import (
+	"bytes"
+	"context"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 
-	"github.com/stretchr/testify/require"
+	"github.com/gomods/athens/pkg/log"
+	"github.com/sirupsen/logrus"
 )
 
-type checkAuthTest struct {
-	name     string
-	user     string
-	pass     string
-	httpUser string
-	httpPass string
-	want     bool
-}
-
-var checkAuthTests = [...]checkAuthTest{
+var basicAuthTests = [...]struct {
+	name   string
+	user   string
+	pass   string
+	path   string
+	logs   string
+	status int
+}{
 	{
-		name:     "success",
-		user:     "athens",
-		pass:     "1234",
-		httpUser: "athens",
-		httpPass: "1234",
-		want:     true,
+		name:   "happy_path",
+		user:   "correctUser",
+		pass:   "correctPass",
+		path:   "/",
+		logs:   "",
+		status: 404,
```

Comment:
Let us change it to use the same endpoint and get a 200, for example `/healthz`. That is far better than a 404. Correct me if I am reading something wrong here @marwan-at-work